### PR TITLE
Consolidate ctors

### PR DESCRIPTION
**Pull request by jbogard**

### CODE REVIEW

File: src/AutoMapper/MapperConfiguration.cs [hunk 1]
```diff
@@ -201,40 +193,41 @@ public TypeMap ResolveTypeMap(Type sourceType, Type destinationType)
...
-                {
-                    typeMap.Seal(this);
-                }
-            }
+            var typeMap = _typeMapPlanCache.GetOrAdd(typePair);
+
+            if (typeMap == null)
+                return null;
+
+            lock (typeMap)
```
Comment (lbargaoanu) [inline src/AutoMapper/MapperConfiguration.cs]: In most cases there is no need for a lock here. That's why those checks were in place. This is wasteful.

**Comment by jbogard** [comment]
Fixes #1985

File: src/AutoMapper/MapperConfiguration.cs (hunk 1 above)
Comment (jbogard) [inline src/AutoMapper/MapperConfiguration.cs]: Which part is wasteful, the lock or the null check?

Comment (lbargaoanu) [inline src/AutoMapper/MapperConfiguration.cs]: The lock, It should be taken only when needed.

Comment (lbargaoanu) [inline src/AutoMapper/MapperConfiguration.cs]: Using Lazy for the map expression is an alternative, but that's needed only when creating maps on the fly so I don't think it's worth it.

Comment (jbogard) [inline src/AutoMapper/MapperConfiguration.cs]: Oh that lock was already there, it just started blowing up because the type map created could be null.

Comment (lbargaoanu) [inline src/AutoMapper/MapperConfiguration.cs]: Yes, but the lock was taken only in certain conditions. You deleted those checks. That's my point :)

**Pull request merged**